### PR TITLE
Fix/2349

### PR DESCRIPTION
**Pull request by MoKob**
Symptomatic fix for https://github.com/Project-OSRM/osrm-backend/issues/2349.

For a full and correct handling, we would require to know about small components at the creation of the edge-based-graph.

### CODE REVIEW

File: features/testbot/via.feature [hunk 1]
```diff
@@ -148,6 +148,24 @@ Feature: Via points
             | a,d,c     | abc,bd,bd,bd,abc,abc |
             | c,d,a     | abc,bd,bd,bd,abc,abc |
 
+    # See issue #2349
```
Comment (TheMarex) [inline features/testbot/via.feature]: I feel I added something like this in https://github.com/Project-OSRM/osrm-backend/pull/1899/files did this get removed by accident again?

**Comment by TheMarex** [comment]
In general I'm okay with this as a hot fix.

We should figure out how big the problem here actually is. My hunch tells me that roads that lead into nothingness will only consist of one compressed edge anyway.

File: features/testbot/via.feature (hunk 1 above)
Comment (MoKob) [inline features/testbot/via.feature]: No, its an addition. Your test is still in there, but is only for exiting oneways, not entering ones.

**Comment by MoKob** [comment]
In general, if your assumption is true, the addition of this fix should handle the problem in general. It is only good as long as the assumption is true. Other cases will still fail.

**Comment by TheMarex** [comment]
@MoKob Yes. Approach here should be:

1. We add a test case that still make this crash (force 2 compressed edges?) and tag with `@bug`
2. We open an issue about this specific case and point to the failing test.
3. We merge this.

Does this sound good to you @MoKob?

**Comment by MoKob** [comment]
Added the testcase

**Pull request merged**